dkg/frost/frost.go

```diff
@@ -0,0 +1,756 @@
+package frost
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+
+	"github.com/bloxapp/ssv-spec/dkg"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/coinbase/kryptology/pkg/dkg/frost"
+	"github.com/coinbase/kryptology/pkg/sharing"
+	ecies "github.com/ecies/go/v2"
+	"github.com/herumi/bls-eth-go-binary/bls"
+	"github.com/pkg/errors"
+)
+
+var thisCurve = curves.BLS12381G1()
+
+func init() {
+	_ = bls.Init(bls.BLS12_381)
+	_ = bls.SetETHmode(bls.EthModeDraft07)
+}
+
+type FROST struct {
+	identifier   dkg.RequestID
+	network      dkg.Network
+	signer       types.DKGSigner
+	storage      dkg.Storage
+	threshold    uint32
+	currentRound DKGRound
+
+	operatorID types.OperatorID
+	operators  []uint32
+	party      *frost.DkgParticipant
+	sessionSK  *ecies.PrivateKey
+
+	msgs           map[DKGRound]map[uint32]*dkg.SignedMessage
+	operatorShares map[uint32]*bls.SecretKey
+}
+
+type DKGRound int
+
+const (
+	Preparation DKGRound = iota + 1
```

Comment:
Why iota + 1?

Reply:
It is to differentiate it from the default zero value of type int. If it starts from `0` i.e `iota` then `New` will set `currentRound` to `Preparation` which starts only after `Start` fn is called.